### ISSUE TITLE
Consider a Client setting to set a consent expiration

### ISSUE THREAD

**Issue by brockallen** [IdentityServer/IdentityServer4#951]
When a user consents we don't set an expiration for the grant. Consider a setting on the Client to set an expiration so they will have to re-consent at some point.

**Comment by chrisowhite** [comment]
@brockallen Poking around I saw somewhere that it was being set based on the authorization code lifetime setting. If you can suggest a reasonable default, I can take this one since I have a PR in for evaluating expiration consistently.

**Comment by brockallen** [comment]
I think this issue is for the "consent" type grant, not the refresh token or other types.

**Comment by chrisowhite** [comment]
Yes, 100%. I was just suggesting I could help with this since I'd been digging around areas concerning this issue while working through the expiration consistency item, and felt comfortable with helping with this enhancement.

**Comment by brockallen** [comment]
Yea, go for it. Thanks.

**Comment by chrisowhite** [comment]
Any suggestion on a reasonable default to apply to the `Client` model?

**Comment by brockallen** [comment]
I was thinking null -- make it optional, and if they set it then it's what the expiration is for user consents. Something along those lines...

**Comment by chrisowhite** [comment]
I started working on this, and it depends on some of the changes I made for #1084 in 833dc0aa58b717578aca78621ef71b826341e83d. Specifically the changes to `DefaultConsentService` and `DefaultGrantStore` along with the relevant test changes.

@brockallen how do you suggest I proceed with this?

**Comment by brockallen** [comment]
Maybe do a PR to your PR?
